code/game/machinery/doors/firedoor.dm

```diff
@@ -82,8 +79,7 @@
 
 /obj/machinery/door/firedoor/LateInitialize()
 	. = ..()
-	GetMergeGroup(merger_id, allowed_types = merger_typecache)
-
+	register_adjacent_turfs(src)
```

Comment:
If you do that, you can drop this as it won't need to exist anymore

Reply:
but I need COMSIG_TURF_EXPOSE and other shit registered

Reply:
register_adjacent_turfs will get called by the signals that building a merge group creates

Reply:
no it wont its not called anywhere outside of Late init

Reply:
oh sorry I uh, I'm dumb.
You should move the GetMergeGroup call into LateInit tho. Not the signal registration mind that's good where it is, it's just important we don't try and generate a mergegroup before everyone sets up their signals and all

Reply:
its in late init